### PR TITLE
Add support for adLibOptions property in executeAdLib

### PR DESCRIPTION
**Pull request by scriptorian**
### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)

Feature

### What is the current behaviour? (You can also link to an open issue here)

The playlist API request executeAdLib allows the client to trigger adLibs that have been created by the blueprints while loading a rundown. The endpoint offers some options for the way these adLibs will be executed though the use of the actionType property that sets a particular trigger mode.

The client discovers the available adLibs and actionTypes by subscribing to the adLibs subscription in the Live Status Gateway (LSG).

### What is the new behaviour (if this is a feature change)?

This change adds a new parameter adLibOptions to the executeAdLib request, allowing the client to modify specific properties of the adLib not available through trigger modes, for example the required duration of a video clip.

The adLibOptions property should conform to the optionsSchema that is provided in the adLibs subscription of the LSG thanks to PR #1221. The changes in combination allow the options property to be passed to the blueprints and validated before being used to modify the adLib execution.

The optionsSchema is expected to support only simple changes to the rundown-created adLibs and the properties will normally be optional. The exception to this is templateAdLibs where some required properties are not known during rundown loading. The options validation in this case will ensure that all necessary properties are available before executing the adLib.

#### Status

- [x] PR is ready to be reviewed.
- [x] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [x] Relevant unit tests have been checked.
- [x] The functionality has been tested by the PR author

#### Affected Areas

* This PR affects the REST API implementation.
* Without blueprint changes there will be no change in behaviour.

### CODE REVIEW

**Comment by nytamin** [comment]
Thank you for this contribution! The NRK team is on vacation at the moment, we'll look at this PR when we're back in August.

File: packages/job-worker/src/playout/adlibAction.ts [hunk 1]
```diff
@@ -102,6 +102,7 @@ export async function executeAdlibActionAndSaveModel(
 		userData: data.userData,
 		triggerMode: data.triggerMode,
 		privateData: adLibActionDoc?.privateData,
+		publicData: data.actionOptions,
```
Comment (jstarpl) [inline packages/job-worker/src/playout/adlibAction.ts:105]: I find this unusual. If I were a blueprint developer, when seeing a `publicData` incoming, I would expect that this is `adLibActionDoc?.publicData`, and not something injected by the user/consumer? Also, what's the difference between this implementation of `publicData` and `userData`? Both are incoming from the user/consumer, so why have them split? Why call it `publicData` if it has nothing to do with the entities `publicData`?

**Review by jstarpl — COMMENTED** [review]
I think I need a bit more explanation on the purpose of this `adLibOptions`/`actionOptions`/`publicData` - the name of this structure changes through the code and leaves me confused. Would the same purpose be achieved by allowing the AdLib Execution REST endpoint to inject a "custom" `userData` on request? How are `adLibOptions`/`actionOptions`/`publicData` different from `userData`?

**Comment by jstarpl** [comment]
We'd like to invite the Contributor and anyone else interested in this PR to a Workshop on Thu 2024-08-22 13:00 - 14:00 to discuss this and it's sibling PRs. Please e-mail me at [jan.starzak@nrk.no](mailto:jan.starzak@nrk.no) for a Microsoft Teams invitation.

**Review by jstarpl — APPROVED** [review]

**Review by jstarpl — APPROVED** [review]

**Pull request merged**